packages/custom-objects-importer/src/main.js

```diff
@@ -0,0 +1,295 @@
+    this.continueOnProblems = options.continueOnProblems || false
+
+    this.logger = {
+      ...silentLogger,
+      ...options.logger,
+    }
+
+    this._initiateSummary()
+  }
+
+  processStream(chunk: Array<CustomObjectDraft>, cb: () => mixed): mixed {
+    this.logger.info(`Starting conversion of ${chunk.length} custom objects`)
+    return this._processBatches(chunk).then(cb)
+  }
+
+  static createBatches(
+    arr: Array<CustomObjectDraft>,
+    batchSize: number
+  ): Array<Array<CustomObjectDraft>> {
+    const tmp = [...arr]
+    let cache = []
+    while (tmp.length) cache = [...cache, tmp.splice(0, batchSize)]
+    return cache
+  }
+
+  static promiseMapSerially(functions: Array<Function>): Promise<void> {
+    return functions.reduce(
+      (
+        promise: Promise<void>,
+        promiseReturningFunction: Function
+      ): Promise<void> =>
+        promise.then((): Promise<void> => promiseReturningFunction()),
+      Promise.resolve()
+    )
+  }
+
+  run(objects: Array<CustomObjectDraft>): Promise<void> {
+    if (objects.length < 1 || !Array.isArray(objects))
+      throw Error(
+        'No objects found, please pass an array with custom objects to the run function'
+      )
+    return this._processBatches(objects)
+  }
+
+  async _processBatches(objects: Array<CustomObjectDraft>): Promise<void> {
+    this.logger.info('Starting Import')
+
+    const batches = CustomObjectsImporter.createBatches(objects, this.batchSize)
+    const uri = CustomObjectsImporter.buildUri(this.apiConfig.projectKey)
+    const existingObjectsRequest = CustomObjectsImporter.buildRequest(
+      uri,
+      'GET'
+    )
+    // skip below because of flow issue with async/await
+    // https://github.com/facebook/flow/issues/5294
+    // todo: remove `FlowFixMe` when above issue is fixed
+    // $FlowFixMe
+    const { body: { results: existingObjects } } = await this.client.execute(
+      existingObjectsRequest
+    )
+
+    const requestsList = batches.map(
+      (newObjects: Array<CustomObjectDraft>): Array<ClientRequest> =>
+        this._createOrUpdateObjects(existingObjects, newObjects)
+    )
+
+    const functionsList = requestsList.map(
+      (requests: Array<ClientRequest>): Function =>
+        this._createPromiseReturningFunction(requests)
+    )
+
+    return CustomObjectsImporter.promiseMapSerially(functionsList)
+  }
+
+  _createPromiseReturningFunction(requests: Array<ClientRequest>): Function {
+    return async (): Promise<void> => {
+      await Promise.all(
+        requests.map((request: ClientRequest): ExecutionResult =>
+          this._executeCreateOrUpdateAction(request)
+        )
+      )
+    }
+  }
+
+  _executeCreateOrUpdateAction(request: ClientRequest): ExecutionResult {
+    let update
+    if (request.body && request.body.update) update = request.body.update
+
+    return this.client
+      .execute(request)
+      .then((): ExecutionResult => {
+        if (update) {
+          this._summary.updated += 1
```

Comment:
ohh.. I didn't think that this would be a number..
maybe a `*Count` name would have been preferable?!